chapters/interface.tex

```diff
@@ -594,10 +594,11 @@ \section{Type Compatible Expressions}\label{type-compatible-expressions}
 \item Binary operators if both operands are of simple types, e.g.\ \lstinline!A + B!.
 Binary operators for other types are only defined for operator records, \cref{overloaded-binary-operations}, and do not necessarily require that the operands are type compatible with each other.
 \end{itemize}
-The resulting type of the expression in case of two type compatible subexpressions \lstinline!A! and \lstinline!B! is defined as follows:
+If the subexpressions satisfy the following restrictions they are called type compatible expressions, and the resulting type of the expression is the type compatible expression.
```

Comment:
I see what you mean.... Maybe just:
```suggestion
If the subexpressions satisfy the following restrictions they are called type compatible expressions, and the type of the resulting expression is also specified below.
```
Then we would also need to update "type compatible expression" to "type of the resulting expression".

Reply:
Yes makes sense.

Reply:
Or perhaps just "type of the expression".

Basically 'resulting' would imply that we do something - but here we started with the expression `{A,B}` so it seems odd to call it the resulting expression. 
We could consider it evaluated and then it is "type of the result".

I also considered 'type of the full expression', to differentiate the type of the subexpressions `A` and `B` from the full expression `{A,B}` - at first I had thought that it may cause confusion due to nested sub-expressions.

Finally I considered adding an example after 'type of the expression' - but I couldn't find a good example-expression. `A+B` doesn't work for records, and `{A,B}` adds an array-layer, and `if x then A else B` is too long to repeat.

I then realized that 'type of the full expression' was the best so far in my opinion.